cmd/tcl/testworkflow-toolkit/commands/artifacts.go

```diff
@@ -94,18 +97,24 @@ func NewArtifactsCmd() *cobra.Command {
 				}
 			}
 
+			var handlerOpts []artifacts.HandlerOpts
 			// Archive
 			if env.CloudEnabled() {
+				ctx, cancel := context.WithTimeout(cmd.Context(), 30*time.Second)
+				defer cancel()
+				client := env.Cloud(ctx)
+				defer client.Close()
+				handlerOpts = append(handlerOpts, artifacts.WithPostProcessor(artifacts.NewJUnitPostProcessor(filesystem.NewOSFileSystem(), client)))
 				if compress != "" {
 					processor = artifacts.NewTarCachedProcessor(compress, compressCachePath)
```

Comment:
I think that it should not be enabled when compression is enabled, as you won't be able to read the file

Reply:
we don't have an actual pipeline like `handler: processor -> uploader -> post-processor`, it is more 2 "parallel" flows like:
```
handler:
  - processor -> uploader
  - post-processor
```
meaning post-processing will be run on the same files which get traversed by the walker, and not on the output of the processor.

That way the junit post-processor can work with any processor.

Now when I think of it, I don't know is post-processor the best name, at least it is current best :D

Reply:
I don't know will we do something else which would require processing on the same files. To be fair, it technically is a pre-processor, as it processes the input files, post-processor would logically process the files outputted by processor.

I like Analyzer tho, we can always rename this interface, it's impact is localized